### PR TITLE
Added onChange event to TextInput in the Form widget

### LINKED ISSUE #709: Add support for FormInput.onChange
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by jkorn8**
## Describe your changes
Changed the TextInput widget and added an onChange property.

## Issue ticket number and link
[Issue #709](https://github.com/EnsembleUI/ensemble-react/issues/709)

Closes #709

## Checklist before requesting a review

- [✅] I have performed a self-review of my code
- [✅] I have added tests
- [✅] I have added a changeset `pnpm changeset add`
- [✅] I have added example usage in the kitchen sink app

### CODE REVIEW

File: packages/runtime/src/widgets/Form/TextInput.tsx [hunk 1]
```diff
@@ -149,12 +162,17 @@ export const TextInput: React.FC<TextInputProps> = (props) => {
     return rulesArray;
   }, [values?.validator, values?.mask]);
 
+  const handleChange = (newValue: string) => {
+    setValue(newValue);
+    actionCallback(newValue);
+  };
```
Comment (evshi) [inline packages/runtime/src/widgets/Form/TextInput.tsx:168]: Can you combine this callback with the one on line 46

**Review by evshi — COMMENTED** [review]
I realize the original discussion was just for TextInput, but curious if you had a chance to take a look at generalizing this for all FormItems?

File: packages/runtime/src/widgets/Form/TextInput.tsx (hunk 1 above)
Comment (jkorn8) [reply packages/runtime/src/widgets/Form/TextInput.tsx:168]: Did not look into it before but will look into it right now!

**Comment by jkorn8** [comment]
> I realize the original discussion was just for TextInput, but curious if you had a chance to take a look at generalizing this for all FormItems?

Did not look into it before but will look into it right now!

File: packages/runtime/src/widgets/Form/MultiSelect.tsx [hunk 2]
```diff
@@ -40,7 +40,7 @@ export type MultiSelectProps = {
   labelKey?: string;
   valueKey?: string;
   items?: Expression<SelectOption[]>;
-  onItemSelect?: EnsembleAction;
```
Comment (evshi) [inline packages/runtime/src/widgets/Form/MultiSelect.tsx:43]: Let's keep this around but mark it as deprecated. So we will prefer onChange if it's there but fallback to onItemSelect.

**Review by evshi — COMMENTED** [review]
Looking good, thanks for cleaning this up!

File: packages/runtime/src/widgets/Form/MultiSelect.tsx [hunk 3]
```diff
@@ -41,6 +41,7 @@ export type MultiSelectProps = {
   valueKey?: string;
   items?: Expression<SelectOption[]>;
   onChange?: EnsembleAction;
+  onItemSelect?: EnsembleAction;
```
Comment (evshi) [inline packages/runtime/src/widgets/Form/MultiSelect.tsx:44]: Can we add a deprecated comment here?

**Review by evshi — APPROVED** [review]

**Pull request merged**